CI: run unittests after build

swt2c opened the pull request:
There are probably going to be failures, but would be good to know where we are.

swt2c committed:
CI: run unittests after build

echoix commented:
For the hanging test, what is the pattern that should be used when running without an user interaction?

In the following, showing a modal dialog has no apparent way of getting closed.

https://github.com/wxWidgets/Phoenix/blob/5abeba2f5dabd6de81c529ac30ca5ae26edd69c3/unittests/test_filedlgcustomize.py#L9-L22

swt2c commented:
@echoix this line should cause the modal to be closed after 250 ms (I think):

```
wx.CallLater(250, dlg.EndModal, wx.ID_OK)
```

I haven't investigated yet why the Windows tests are hanging - is this the reason?

echoix commented:
If it were to be a deadlock caused by multiprocessing and the startup method, we might've been able to see something similar on macOS too. Was the job retried and jams at the same place?

echoix commented:
Last time I debugged something similar, I had to go by bisection to find what test where actually causing to jam, as it's not always the ones we see. If I remember correctly, there was also something to do with the fact that the shell used in the CI was cmd, and is quite limited on error reporting. The test was already failed, but the error didn't display yet, waiting for some input on some stream, and then times out, and further gets force killed (when seeing the ^C).

It's probably not as worse as it looks like.

echoix commented:
I've done some more runs on a PR against my fork: https://github.com/echoix/Phoenix/pull/21
I tried to see if it was the nested class inside a function that was problematic, but no, it wasn't that : (https://github.com/echoix/Phoenix/pull/22).



So I tried skipping the hanging tests to see what else there was, or if there was a pattern.
First I updated from main to make sure nothing changed, just in case.
After skipping the single test_filedlgcustomize.py test, for windows with python < 3.13 (as 3.13 works), there was another file that was hanging, unittests/test_richmsgdlg.py. I skipped all of these 3 tests, as I didn't see after killing one dot, so I didn't know if it was one or more tests that were hanging (not with 3.13 on windows again).

I added other python versions for windows x64, and nothing else failed or hung.
Now we can analyze the pattern of what is common in both of the skipped files, and why is it working on 3.13.

Once the CI is finished I'll try to push again to narrow what python versions hang, and also see which ones of the three richmsgdlg_Tests fail. Three of them use dlg.ShowModal() and wx.CallLater() too.
But a file like /unittests/test_msgdlg.py also has similar patterns and it didn't cause a hang.

swt2c committed:
Remove numpy Python version restriction

This was originally added when numpy didn't support Python 3.12, but
that has long since been resolved.

swt2c committed:
Fail build.py if unittests fail

swt2c committed:
Try enabling pytest_timeout

echoix commented:
There's a GH outage

echoix commented:
They are only processing 15% of webhooks now

swt2c commented:
> There's a GH outage

Yeah I found that after nothing I tried was triggering the actions :(

swt2c committed:
TEMP HACK INSTALL PREBUILT WHEELS

swt2c committed:
Fix test_filedlgcustomize on MSW

EndModal doesn't work because wx.FileDialog isn't a real modal dialog on MSW.

swt2c committed:
Avoid using Icon property on empty IconBundle in test_iconbndl

This seems to have not crashed in 3.2, but seems to now.

swt2c committed:
Remove deprecated methods from DataViewEvent test in test_dataview

swt2c committed:
Remove un-documented TL_STYLE_MASK from test_treelist

swt2c committed:
Remove deprecated constants from test_stc

swt2c committed:
Remove wx.Bitmap.SetSize() test as this is no longer supported in wx 3.3

SetHeight and SetWidth have been removed, so SetSize has been removed
from wxPython.

swt2c committed:
Fix test_richmsgdlg on MSW

The same issue as with test_filedlgcustomize applies here as well.

swt2c commented:
Took quite a bit of work but I think I finally got this sorted out (found a lot of test bugs, but also some real code bugs as well) so we can have working unittests in CI :)

echoix commented:
> Took quite a bit of work but I think I finally got this sorted out (found a lot of test bugs, but also some real code bugs as well) so we can have working unittests in CI :)

I found you were quite persevering when you continued to try again and again. I would’ve started with the ones that work, and then work on the unfixed ones. Or let an agent do all the testing/trial and error of the failures and after that take a look at what to actually do.

swt2c commented:
> I found you were quite persevering when you continued to try again and again. I would’ve started with the ones that work, and then work on the unfixed ones. Or let an agent do all the testing/trial and error of the failures and after that take a look at what to actually do.

I thought it was a worthwhile task to get all tests passing now, as it found a lot of issues that were broken in the move to wxWidgets 3.3.
